master/buildbot/locks.py

```diff
@@ -294,7 +309,7 @@ def __init__(self, lockid, mode, _skipChecks=False):
             assert mode in ['counting', 'exclusive']
 
 
-class BaseLockId(util.ComparableMixin):
+class BaseLockId:
```

Comment:
Where do we do that? I think we need a test to prevent similar changes from breaking buildbot.

Reply:
ComparableMixin allows to specify which attributes are important for comparaison with ==
It is recursive so that if an item specifies locks in its list of compare_attrs, then it will detect a change based on name change or maxCount. If you are not using this mixin, then == by default compare the id(), then it will always declare to instance different. Please have a look at the source code for more understanding. It is a very neat feature of buildbot.

Reply:
Reverted the commit.